dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/udp/UDPUtils.java

```diff
@@ -0,0 +1,256 @@
+ * Copyright (c) 2021, 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.netty.internal.udp;
+
+import java.net.Inet6Address;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.socket.nio.NioDatagramChannel;
+import io.openliberty.netty.internal.BootstrapExtended;
+import io.openliberty.netty.internal.ConfigConstants;
+import io.openliberty.netty.internal.exception.NettyException;
+import io.openliberty.netty.internal.impl.NettyFrameworkImpl;
+import io.openliberty.netty.internal.impl.NettyConstants;
+
+public class UDPUtils {
+
+    private static final TraceComponent tc = Tr.register(UDPUtils.class, UDPMessageConstants.NETTY_TRACE_NAME,
+            UDPMessageConstants.UDP_BUNDLE);
+
+    /**
+     * Create a {@link BootstrapExtended} for inbound UDP channels
+     * 
+     * @param framework
+     * @param options
+     * @return
+     * @throws NettyException
+     */
+    public static BootstrapExtended createUDPBootstrap(NettyFrameworkImpl framework, Map<String, Object> options)
+            throws NettyException {
+        return create(framework, new UDPConfigurationImpl(options, true));
+    }
+
+    /**
+     * Create a {@link BootstrapExtended} for outbound channels
+     * 
+     * @param framework
+     * @param options
+     * @return
+     * @throws NettyException
+     */
+    public static BootstrapExtended createUDPBootstrapOutbound(NettyFrameworkImpl framework,
+            Map<String, Object> options) throws NettyException {
+        return create(framework, new UDPConfigurationImpl(options, false));
+    }
+
+    private static BootstrapExtended create(NettyFrameworkImpl framework, UDPConfigurationImpl config) {
+        BootstrapExtended bs = new BootstrapExtended();
+        bs.applyConfiguration(config);
+        bs.group(framework.getParentGroup());
+        bs.channel(NioDatagramChannel.class);
+        return bs;
+    }
+
+    private static ChannelFuture bind(NettyFrameworkImpl framework, BootstrapExtended bootstrap, String inetHost,
+            int inetPort, final int retryCount, final int retryDelay, ChannelFutureListener bindListener) {
+        ChannelFuture bindFuture = bootstrap.bind(inetHost, inetPort);
```

Comment:
Interesting that we use inetHost 'as is' * passing it into bind() and **then** we set it to INADDR_ANY.
So methods have to be able to deal with "*" and/or INADDR_ANY or just some or just bind()?
It does not need to change, but sometimes it is better to use one of two alternatives everywhere,
getting rid of the either/or as soon as possible and being consistent afterwards.

Reply:
We may want to follow up on this on https://github.com/OpenLiberty/open-liberty/issues/18520